dateparser/languages/language.py

```diff
@@ -24,40 +26,51 @@ def __init__(self, shortname, language_info):
             if isinstance(value, int):
                 simplification[key] = str(value)
 
+        self._cached = self
```

Comment:
But now original methods are never called, they all proxying to `self._cached`. Is there a point?
How do you feel about instead of implementing this on a `Language` class level go level above and try to implement this on `LanguageDataLoader` level, or `BaseLanguageDetector` level or even `DateDataParser` level?